### PR TITLE
Support for android MNC

### PR DESCRIPTION
**Pull request by teobugslayer**
Whitelist MNC SDK
change all vars to lets
When building with MNC SDK, change minSdkVersion to be MNC, required for succesful build
Use latest common lib to support MNC emulator

See https://github.com/NativeScript/nativescript-cli/issues/507

### CODE REVIEW

File: lib/services/android-project-service.ts [hunk 1]
```diff
@@ -111,27 +110,35 @@ class AndroidProjectService implements IPlatformProjectService {
...
 			this.runAndroidUpdate(projectRoot, targetApi).wait();
+			this.adjustMinSdk(projectRoot);
 		}).future<void>()();
 	}
 
+	private adjustMinSdk(projectRoot: string): void {
+		let manifestPath = path.join(projectRoot, "AndroidManifest.xml");
+		let apiLevel = this.getTarget(projectRoot).wait().split('-')[1];
+		if (apiLevel === "MNC") { // MNC SDK requires that minSdkVersion is set to "MNC"
+			shell.sed('-i', /android:minSdkVersion=".*"/, `android:minSdkVersion="${apiLevel}"`, manifestPath);
```
Comment (rosen-vladimirov) [inline lib/services/android-project-service.ts]: should you include ? in the regex. If you have line android:minSdkVersion="L" android:smthElse="Value", the match will be everything up to Value".

File: lib/services/android-project-service.ts [hunk 2]
```diff
@@ -179,25 +184,25 @@ class AndroidProjectService implements IPlatformProjectService {
 		this.$logger.info("Copying %s", projDir);
 		shell.cp("-Rf", projDir, destDir);
 
-		var targetDir = path.join(destDir, path.basename(projDir));
+		let targetDir = path.join(destDir, path.basename(projDir));
 		// TODO: parametrize targetSdk
-		var targetSdk = "android-17";
+		let targetSdk = this.targetApi || "android-17";
```
Comment (rosen-vladimirov) [inline lib/services/android-project-service.ts]: is it possible to have `this.targetApi` still unpopulated? Shouldn't we call this.getTarget (, just to make sure targetApi has correct value?
As far as I see this.targetApi is populated only in this.getTarget method:
https://github.com/NativeScript/nativescript-cli/blob/totev/mnc/lib/services/android-project-service.ts#L379

**Comment by rosen-vladimirov** [comment]
:+1:  after green build

File: lib/services/android-project-service.ts [hunk 3]
```diff
@@ -179,25 +184,25 @@ class AndroidProjectService implements IPlatformProjectService {
 		this.$logger.info("Copying %s", projDir);
 		shell.cp("-Rf", projDir, destDir);
 
-		var targetDir = path.join(destDir, path.basename(projDir));
+		let targetDir = path.join(destDir, path.basename(projDir));
 		// TODO: parametrize targetSdk
-		var targetSdk = "android-17";
+		let targetSdk = this.getTarget(projDir).wait() || "android-17";
```
Comment (Fatme) [inline lib/services/android-project-service.ts]: Why do we need this change here? This will update only `project.properties` file

Comment (teobugslayer) [inline lib/services/android-project-service.ts]: After personal discussion we decided to fix this in a separate PR.

**Comment by Fatme** [comment]
:+1:

**Pull request merged**